src/main/groovy/com/google/protobuf/gradle/ToolsLocator.groovy

```diff
@@ -108,4 +108,13 @@ class ToolsLocator {
       }
     }
   }
+
+  static List<String> artifactParts(String artifactCoordinate) {
+    final validName = '[+-.0-9A-Z_a-z]+'
+
+    ((artifactCoordinate =~ /(${validName}):(${validName}):(${validName})(:${validName})?(@${validName})?/)[0]
```

Comment:
`codeNarcMain` and `codeNarcTest` are the tasks to run. The issue I was having had to do with the rules having been updated and not merged into my branch (so the tasks would pass locally but not in the CI). After merging upstream master, I'm able to reproduce the errors.

I'll update the code with something similar to what you have above.